### PR TITLE
Add register-service script for testnet ServiceRegistry

### LINKED ISSUE #447: Register service at ServiceRegistry
## Description

**As Josh I want to be able to**

register my services in the ServiceRegistry

**So that I can**

be fund by clients.

## Acceptance Criteria

- [ ] Have a script that registers the services in the ServiceRegistry.

### PR DESCRIPTION
**Pull request by palango**
Fixes #447

I need to write tests for this too....

### CODE REVIEW

File: src/raiden_libs/cli.py [hunk 1]
```diff
@@ -132,10 +133,11 @@ def blockchain_options(contracts: List[str]) -> Callable:
     ]
 
     arg_for_contract = {
-        CONTRACT_TOKEN_NETWORK_REGISTRY: "token-network-registry",
+        CONTRACT_TOKEN_NETWORK_REGISTRY: "token-network-registry-contract",
```
Comment (karlb) [inline src/raiden_libs/cli.py]: This needs to be updated in the docs.

File: src/raiden_libs/register_service.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        service_address=service_address,
+        current_deposit=current_deposit,
+        current_url=current_url,
+    )
+
+    # Register if not yet done
+    if current_deposit <= 0:
+        log.info("Address not registered in ServiceRegistry")
+
+        custom_token_address = contracts[CONTRACT_USER_DEPOSIT].functions.token().call()
```
Comment (karlb) [inline src/raiden_libs/register_service.py]: Custom token? Isn't this assumed to be RDN? For the test contract deploys it isn't, but `custom_token` still sounds very vague.

File: src/raiden_libs/register_service.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            abi=CONTRACT_MANAGER.get_contract_abi(CONTRACT_CUSTOM_TOKEN),
+        )
+
+        # Check current token balance
+        account_balance = custom_token_contract.functions.balanceOf(service_address).call()
+        log.info("Current account balance", balance=account_balance, desired_deposit=deposit)
+
+        # mint token if necessary
+        if account_balance < deposit:
+            run_task(
```
Comment (karlb) [inline src/raiden_libs/register_service.py]: `run_task` sounds like concurrent stuff is happening. Maybe just call it something like `checked_transact`?

File: src/raiden_libs/register_service.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+@common_options("register_service")
+def main(
+    private_key: str,
+    state_db: str,
+    web3: Web3,
+    contracts: Dict[str, Contract],
+    start_block: BlockNumber,
+    deposit: TokenAmount,
+    service_url: str,
+) -> None:
```
Comment (karlb) [inline src/raiden_libs/register_service.py]: If we add a docstring, it should show up in the cli. We could inform the user that used key define be both the payer as well as the service address.

File: src/raiden_libs/register_service.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                function_call=custom_token_contract.functions.mint(deposit),
+                task_name="Minting new Test RDN tokens",
+            )
+
+            account_balance = custom_token_contract.functions.balanceOf(service_address).call()
+            log.info("Updated account balance", balance=account_balance, desired_deposit=deposit)
+
+        current_allowance = custom_token_contract.functions.allowance(
+            service_address, service_registry_contract.address
+        ).call()
```
Comment (karlb) [inline src/raiden_libs/register_service.py]: Do we need the allowance check? Usually there should be now allowance left, and if there is, setting it again won't cause more problems than the wasted gas.

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (palango) [reply src/raiden_libs/cli.py]: Do you think the change is worthwhile or should I undo it? It just didn't seem consistent

Comment (karlb) [reply src/raiden_libs/cli.py]: When I added it, I intentionally kept the name to keep it the same as for the client. But now neither have to be entered explicitly, usually. I don't mind either way.

File: src/raiden_libs/register_service.py (hunk 5 above)
Comment (palango) [reply src/raiden_libs/register_service.py]: You mean to get rid of the check and call `allow` unconditionally?

Comment (karlb) [reply src/raiden_libs/register_service.py]: `approve` unconditionally, yes. From my understanding, that's what will happen in > 99% of the cases, anyway.

File: src/raiden_libs/cli.py (hunk 1 above)
Comment (palango) [reply src/raiden_libs/cli.py]: Let's do it in a different PR

**Comment by palango** [comment]
@karlb Please have another look.

File: src/raiden_libs/register_service.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    web3: Web3,
+    contracts: Dict[str, Contract],
+    start_block: BlockNumber,  # pylint: disable=unused-argument
+    deposit: TokenAmount,
+    service_url: str,
+) -> None:
+    """
+    Registers the address of a service deployment with the `ServiceRegistry`.
+
+    It also set or updates the URL of the services deployment.
```
Comment (karlb) [inline src/raiden_libs/register_service.py]:
```suggestion
    It also sets or updates the URL of the services deployment.
```

Maybe add one more sentence to say that the `private_key`s address will be registered. Or is that obvious?

**Pull request merged**